docs/en/reference/dql-doctrine-query-language.rst

```diff
@@ -49,9 +49,12 @@ SELECT queries
 DQL SELECT clause
 ~~~~~~~~~~~~~~~~~
 
-The select clause of a DQL query specifies what appears in the
-query result. The composition of all the expressions in the select
-clause also influences the nature of the query result.
+The SELECT clause of a DQL query specifies what gets hydrated in
+the query result. You are always returned usable objects, but any
+associated objects not included in the SELECT clause will be
+proxies (ie. unhydrated).  They get hydrated by Doctrine when
```

Comment:
> Either way I should delete this part. Except I'm not sure it's clear that including things in the SELECT clause is what hydrates them. The discussion of fetch joins at the top is attempting to do that.

As long as it's documented somewhere else, heh.